lib/bolt/node/winrm.rb

```diff
@@ -9,9 +9,14 @@ def initialize(host, port, user, password, shell: :powershell, **kwargs)
 
       @shell = shell
       @endpoint = "http://#{host}:#{port}/wsman"
-      @connection = ::WinRM::Connection.new(endpoint: @endpoint,
-                                            user: @user,
-                                            password: @password)
+      options = { endpoint: @endpoint,
+                  user: @user,
+                  password: @password }
+      if @timeout then
+        options[:receive_timeout] = @timeout
```

Comment:
So far this is not working as expected... working to further investigate WinRM protocol and the gems behavior.